src/odin/commands/status.rs

```diff
@@ -2,48 +2,46 @@ use crate::server::ServerInfo;
 use crate::utils::environment::fetch_var;
 use crate::utils::parse_arg_variable;
 use clap::ArgMatches;
-use log::info;
+use log::{error, info};
 use std::env;
+use std::net::{AddrParseError, SocketAddrV4};
+use std::process::exit;
+use std::str::FromStr;
 
-pub fn fetch_public_address() -> String {
+pub fn fetch_public_address() -> Result<SocketAddrV4, AddrParseError> {
   let current_ip = env::var("ADDRESS").unwrap_or_else(|_| {
     reqwest::blocking::get("https://api.ipify.org")
       .unwrap()
       .text()
       .unwrap()
   });
   let current_port: u16 = fetch_var("PORT", "2456").parse().unwrap();
-  format!("{:?}:{:?}", current_ip, current_port + 1)
+  SocketAddrV4::from_str(&format!("{}:{}", current_ip, current_port + 1))
 }
 
-fn parse_address(args: &ArgMatches) -> String {
+fn parse_address(args: &ArgMatches) -> Result<SocketAddrV4, AddrParseError> {
   let has_address = args.is_present("address");
   if has_address {
-    parse_arg_variable(args, "address", "".to_string())
+    SocketAddrV4::from_str(&parse_arg_variable(args, "address", "".to_string()))
   } else {
     fetch_public_address()
   }
 }
 
-fn pretty_print(status: ServerInfo) {
-  info!("Name: {}", status.name);
-  info!("Players: {}/{}", status.players, status.max_players);
-  info!("Map: {}", status.map);
-  let bepinex = status.bepinex;
-  info!("BepInEx Enabled: {}", bepinex.enabled);
-  if bepinex.enabled {
-    let mods: Vec<String> = bepinex.mods.iter().map(|m| String::from(&m.name)).collect();
-    info!("BepInEx Mods: {}", mods.join(", "))
-  }
-}
-
 pub fn invoke(args: &ArgMatches) {
   let output_json = args.is_present("json");
-  let address = parse_address(&args);
+  let address = match parse_address(&args) {
+    Ok(addr) => addr.to_string(),
+    Err(_) => {
+      let addr = fetch_var("ADDRESS", args.value_of("address").unwrap());
+      error!("Failed to parse supplied address! {}", addr);
+      exit(1)
+    }
+  };
```

Comment:
[`a2s::A2SClient.info`](https://docs.rs/a2s/0.2.3/a2s/struct.A2SClient.html#method.info) takes a `ToSocketAddrs` so you can use the `SocketAddrV4` all the way through to querying the info instead of converting it to a string here. This would make it so you only have to handle the error case

```rust
let address = parse_address(&args).unwrap_or_else(|_| {
    // The stuff in the `Err` arm
    ...
});
```

Reply:
good call